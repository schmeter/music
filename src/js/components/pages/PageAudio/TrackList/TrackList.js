import React from 'react';
import PropTypes from 'prop-types';

import Track from '../Track';

class TrackList extends React.Component {
    render() {
        const { trackList } = this.props;

        return (
            <div className="table tracks">
                {trackList.map((track) => (
                    <Track
                        key={track.path}
                        className="table-row"
                        file={track}
                    />
                ))}
            </div>
        );
    }
}

TrackList.propTypes = {
    trackList: PropTypes.array.isRequired
};

export default TrackList;
