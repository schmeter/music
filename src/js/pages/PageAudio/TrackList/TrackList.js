import React from 'react';
import PropTypes from 'prop-types';

import Track from '../Track';

const TrackList = ({
  tracks,
}) => (
  <div className="table tracks">
    {tracks.map(track => (
      <Track
        key={track.path}
        className="table-row"
        track={track}
      />
    ))}
  </div>
);

TrackList.propTypes = {
  tracks: PropTypes.array.isRequired,
};

export default TrackList;
