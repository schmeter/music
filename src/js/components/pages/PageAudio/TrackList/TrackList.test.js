import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TrackList from './TrackList';

jest.mock('../Track', () => () => <div />);

it('renders correctly', () => {
    const props = {
        tracks: [{
            path: 'test',
        }, {
            path: 'test2',
        }],
    };
    const component = shallow(<TrackList {...props} />);

    expect(toJson(component)).toMatchSnapshot();
});

