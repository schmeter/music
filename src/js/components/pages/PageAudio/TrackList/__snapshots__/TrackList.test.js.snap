// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList renders correctly 1`] = `
<div
  className="table tracks"
>
  <Component
    className="table-row"
    key="test"
    track={
      Object {
        "path": "test",
      }
    }
  />
  <Component
    className="table-row"
    key="test2"
    track={
      Object {
        "path": "test2",
      }
    }
  />
</div>
`;
